<launch>
	<arg name="namespace" default="yk_creator" />

	<!-- Launch femto_bolt.launch from vibro_tactile_toolbox package -->
	<include
		file="$(find vibro_tactile_toolbox)/launch/femto_bolt.launch">
		<arg name="camera_name" value="$(arg namespace)/side_camera"/>
		<arg name="serial_number" value="CL8FC3100XE"/>
	</include>

	<!-- Launch rs_aligned_depth.launch from realsense2_camera package -->
	<include
		file="$(find vibro_tactile_toolbox)/launch/realsense.launch">
		<arg name="namespace" value="$(arg namespace)"/>
		<arg name="camera_name" value="wrist_camera"/>
		<arg name="serial_number" value=""/>
	</include>

	<group ns="$(arg namespace)">
		<node name="sounddevice_ros_node" pkg="sounddevice_ros" type="sounddevice_ros_publisher_node.py" output="screen" args="-d UMC404HD 192k: USB Audio -c 4">
		</node>

		<!-- Start compressed_image_republisher node -->
		<node name="compressed_image_cropped_republisher" pkg="vibro_tactile_toolbox" type="compressed_image_cropped_republisher.py" output="screen">
			<param name="compressed_image_sub_topic_name" value="side_camera/color/image_raw/compressed" />
			<param name="compressed_image_pub_topic_name" value="side_camera/color/image_cropped/compressed" />
			<param name="x_offset" value="200" />
			<param name="y_offset" value="1950" />
			<param name="x_size" value="1280" />
			<param name="y_size" value="720" />
			<param name="rotation_direction" value="clockwise" />
		</node>

	</group>
</launch>
